network-api/networkapi/wagtailpages/models.py

```diff
@@ -486,6 +487,43 @@ def get_context(self, request):
         return get_page_tree_information(self, context)
 
 
+class BanneredCampaignPage(PrimaryPage):
+    """
+    title, header, intro, and body are inherited from PrimaryPage
+    """
+
+    # Note that this is a different related_name, as the `page`
+    # name is already taken as back-referenced to CampaignPage.
+    cta = models.ForeignKey(
+        'Petition',
+        related_name='bcpage',
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+        help_text='Choose existing or create new sign-up form'
+    )
+
+    content_panels = PrimaryPage.content_panels + [
+        SnippetChooserPanel('cta')
+    ]
+
+    parent_page_types = [
+        'HomePage',
+        'MiniSiteNameSpace',
+        'BanneredCampaignPage',
+    ]
+
+    subpage_types = [
+        'BanneredCampaignPage',
```

Comment:
This may not be true for long, but it's fine to start with.

Reply:
I want to audit this in general, because we're mixing parent and subpage lists and it's really hard to control what goes where when parents get to say who their children are AND children get to say who their parents are allowed to be.

Reply:
https://github.com/mozilla/foundation.mozilla.org/issues/2734